Alert engine for combined ATOF/AHDC reconstruction

whit2333 opened the pull request:
- Moved ATOF Track projections to ALERTEngine since it will need a track
- Added many todo comments in code.
- Cleaned up various if statements for clarity.
- moved services to proper project directory structure that is picked up by javadocs

whit2333 committed:
Adding CI stuff that is different from the main dev branch

whit2333 committed:
petiroc timestamps

whit2333 committed:
Merge branch 'development' of github.com:JeffersonLab/coatjava into development

whit2333 committed:
modified:   ahdc/AI/Model.java
	modified:   ahdc/Cluster/ClusterFinder.java
	modified:   ahdc/Distance/Distance.java
	modified:   ahdc/HelixFit/HelixFitJava.java
	modified:   ahdc/HelixFit/HelixFitObject.java
	modified:   ahdc/PreCluster/PreCluster.java
	modified:   service/AHDCEngine.java

whit2333 committed:
Added ALERTEngine and AHDC documentation.

whit2333 committed:
Please enter the commit message for your changes. Lines starting
	modified:   reconstruction/alert/src/main/java/org/jlab/rec/service/ALERTEngine.java

whit2333 committed:
modified:   reconstruction/alert/src/main/java/org/jlab/rec/service/ALERTEngine.java

whit2333 committed:
Cleaned up the track projection code

whit2333 committed:
modified:   reconstruction/alert/src/main/java/org/jlab/rec/service/ATOFEngine.java

whit2333 committed:
modified:   reconstruction/alert/src/main/java/org/jlab/rec/service/ALERTEngine.java
	modified:   reconstruction/alert/src/main/java/org/jlab/rec/service/ATOFEngine.java

whit2333 committed:
Merge remote-tracking branch 'origin/development' into ALERT_Engine

baltzell commented:
I suspect people might prefer to leave the old ones intact for now, maybe just suffix their names to designate they're the old ones.

whit2333 committed:
Merge remote-tracking branch 'origin/development' into ALERT_Engine

whit2333 commented:
> I suspect people might prefer to leave the old ones intact for now, maybe just suffix their names to designate they're the old ones.

Not sure what you are saying? What the old Engines are  still there and used. The only change is I moved the Track projections into ALERTEngine where it is more appropriate.

baltzell commented:
Yes, the old ones'names are still there and used, but their functionality is changed, and their functionality is no longer easily accessible.  In the short term, it'd probably be better to keep their functionality easily accessible, and one easy way to do that is by renaming those old classes with a suffix, rather than replacing them, simultaneously designating that they'll be going away in the near futur.

whit2333 commented:
I still don't understand. All I have done is add a new engine called ALERTEngine. None of the functionality changed. I can move the projection back to ATOFEngine but the next PR will move it out of there....

whit2333 committed:
modified:   service/AHDCEngine.java
	modified:   service/ALERTEngine.java
	modified:   service/ATOFEngine.java

whit2333 committed:
modified:   service/ALERTEngine.java

whit2333 committed:
new file:   alert/banks/RecoBankWriter.java
	modified:   service/ALERTEngine.java

whit2333 committed:
modified:   .gitlab-ci.yml

whit2333 committed:
modified:   .gitlab-ci.yml

whit2333 committed:
modified:   common-tools/clas-geometry/src/main/java/org/jlab/geom/base/Component.java

whit2333 committed:
modified:   build-javadocs.sh

whit2333 committed:
modified:   pom.xml

whit2333 committed:
modified:   .gitlab-ci.yml

whit2333 committed:
modified:   build-javadocs.sh

whit2333 committed:
modified:   .gitlab-ci.yml

whit2333 committed:
Fix so javadocs works.
	modified:   ../../../../../../../../common-tools/clas-geometry/src/main/java/org/jlab/geom/base/Component.java
	renamed:    ../rec/service/AHDCEngine.java -> ahdc/AHDCEngine.java
	renamed:    ../rec/service/ALERTEngine.java -> alert/ALERTEngine.java
	renamed:    ../rec/service/ATOFEngine.java -> atof/ATOFEngine.java
